external/storm-jdbc/src/main/java/org/apache/storm/jdbc/common/JDBCClient.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.jdbc.common;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Lists;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.sql.Date;
+import java.util.*;
+
+public class JDBCClient {
+    private static final Logger LOG = LoggerFactory.getLogger(JDBCClient.class);
+
+    private HikariDataSource dataSource;
+
+    public JDBCClient(Map<String, Object> map) {
+        Properties properties = new Properties();
+        properties.putAll(map);
+        HikariConfig config = new HikariConfig(properties);
+        this.dataSource = new HikariDataSource(config);
```

Comment:
So for each bolt, how many connections do you advise to be in the pool? Given each bolt has one thread then I would say 1. But then it would loose time reconnecting each tuple that may have caused some kind of error. So then I would be inclined to say 2 connections, but then for 10 bolts, I would need 20 connections. What is the recommended value you use in production?